x-pack/plugins/endpoint/server/plugin.ts

```diff
@@ -21,9 +29,11 @@ declare module 'kibana/server' {
 
 export class EndpointPlugin implements Plugin {
   private readonly logger: Logger;
+  private readonly logFactory: LoggerFactory;
   private clusterClient?: IClusterClient;
   constructor(private readonly initializerContext: PluginInitializerContext) {
     this.logger = this.initializerContext.logger.get();
+    this.logFactory = this.initializerContext.logger;
```

Comment:
If you are going to go with the factory then make it consistent. Are you not able to get a logger with the logger object?